mobile/ethclient.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Contains a wrapper for the Ethereum client.
+
+package geth
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// EthereumClient provides access to the Ethereum APIs.
+type EthereumClient struct {
+	client *ethclient.Client
+}
+
+// GetBlockByHash returns the given full block.
+func (ec *EthereumClient) GetBlockByHash(ctx *Context, hash *Hash) (*Block, error) {
+	block, err := ec.client.BlockByHash(ctx.context, hash.hash)
+	return &Block{block}, err
+}
```

Comment:
How would that look like in Java? Would it entail having a single context per ethereum client? Also would it not disable using a single client for multiple things?